syntax="proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.example";
option java_outer_classname = "Example";
option objc_class_prefix = "EXM";

package example;

//Interface for Customer To Bank Transaction
service CustomerTransaction {
  // A simple RPC.
  //
  // Sends a CTransaction Type
  //
  // A CResponse is returned with with success or failure or balance of the account.
    rpc Query(CTransaction) returns(CResponse) {}
    rpc Deposit(CTransaction) returns(CResponse) {}
    rpc Withdraw(CTransaction) returns(CResponse) {}
    rpc Propagate_Deposit(BTransaction) returns(BResponse) {}
    rpc Propagate_Withdraw(BTransaction) returns(BResponse) {}
}

message CTransaction {
    int32 cust_id = 1;
    int32 tran_id = 2;
    string interface = 3;
    float money = 4;
}

message CResponse {
    int32 cust_id = 1;
    int32 tran_id = 2;
    string interface = 3;
    string result = 4;
}


message BTransaction {
    int32 cust_id = 1;
    int32 tran_id = 2;
    int32 src_branch_id = 3;
    string interface = 4;
    float money = 5;
}

message BResponse {
    int32 cust_id = 1;
    int32 tran_id = 2;
    string interface = 3;
    float money = 4;
    bool status = 5;
}
